### PR TITLE
[2.0.x] Update fonts data and scripts

### PR DESCRIPTION
**Pull request by yhfudev**
[Concise Diff](10417/files?w=1)

### Description
This PR is to update the font data. The new font file `marlin-6x12-2.bdf` which is base on the X windows 6x12 bitmap font, it was merged with jp-kana and Marlin symbols in the folder `buildroot/share/fonts`. So the file can be used to update the glyph by the font maintainer.

The font file `marlin-6x12-3.bdf` is for generating language data, it contains the glyph from `marlin-6x12-2.bdf` and the other missing glyph were from WQY 9pt bitmap. The file can be updated once the `marlin-6x12-2.bdf` is updated with new glyph.

History of `marlin-6x12-?.bdf` files:
```
marlin-6x12-1.bdf # base on X Window 6x12 bitmap which covers most of western languages; and Marlin_symbols.fon
marlin-6x12-2.bdf # based on marlin-6x12-1.bdf, and added ISO10646_Kana.fon(30A0-30FF), it obsoleted all of the old files:
  - ISO10646_CZ.fon("ÁÉÍÓÚÝáéíóúýČčĎďĚěŇňŘřŠšŤťŮůŽž")
  - ISO10646_SK.fon('ĹĺĽľ')
  - ISO10646-4_Greek.fon(0384-03FF)
  - ISO10646-5_Cyrillic.fon(0400-045F)
marlin-6x12-3.bdf # merged with marlin-6x12-2.bdf and WQY 9pt bitmap
```

The other files are obsoleted and you may put it to a sub-folder such as 'old' for any reason:
```
HD44780_C.fon # the glyph dump from HD44780 Cyrillic
HD44780_J.fon # the glyph dump from HD44780 Japanese
HD44780_W.fon # the glyph dump from HD44780 Western
ISO10646_Kana.fon
Marlin_symbols.fon
```
the following font should be removed:
```
ISO10646-1-tr.fon # almost the same as `ISO10646-1.fon`, added with non-unicode glyph such as 'ĞğşŞ', which should be merged to location 011E, 011F, 015E and 015F
ISO10646_CN.fon
ISO10646_SK.fon
ISO10646_CZ.fon # base on ISO10646-1-tr.fon, added non-unicode glyph such as 'ÁÉÍÓÚÝáéíóúýČčĎďĚěŇňŘřŠšŤťŮůŽž' at the position 0x80-0x9D in the font file
ISO10646-1.fon # almost the same as HD44780_W.fon, except it's removed non-unicode glyph.
ISO10646-4_Greek.fon
ISO10646-5_Cyrillic.fon
```

This PR also remove some unused macro as well.

### Benefits
Update the font.

### Related Issues

#10408, #10213,#10429

### CODE REVIEW

**Comment by rmoravcik** [comment]
That was fast. Thank you very much. I will test it tomorrow morning. I have one Arduino + LCD at work for testing of Marlin

**Comment by thinkyhead** [comment]
Thanks for this update! I will also make the recommended changes soon.

**Comment by yhfudev** [comment]
@rmoravcik 
This patch only update the missing font data for sk, the problem of different size of font still not fixed. There's various size of glyph in the font data because languages has their own font/size before merging.

**Comment by thinkyhead** [comment]
@yhfudev — I've added a commit to make the suggested changes. There's no need to move anything into a "`fonts/old`" folder if they won't be needed in the future. Do you want to make any additional changes before this is merged?

**Comment by yhfudev** [comment]
@thinkyhead 
Not yet.
There still have problem with the merged font. such as the different font size, different font style etc.
#10429

**Comment by rmoravcik** [comment]
Ok, I just tested slovak characters and it's possible to display all of them now.

**Comment by thinkyhead** [comment]
@yhfudev — Whenever it's ready. There's not a big hurry.

**Comment by rmoravcik** [comment]
I just tested english, czech and slovak and they looks ok. But as I wrote in #1042,9 I had to change value of 

`#define TALL_FONT_CORRECTION 1` 

to

`#define TALL_FONT_CORRECTION 0` 

in order to fit 5 menu items to screen without scrolling...

File: Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h [hunk 1]
```diff
@@ -26,19 +26,20 @@
...
-  Pure Font   ascent = 7 descent=-1
-  X Font      ascent = 7 descent=-1
+  Pure Font   ascent = 7 descent=-2
+  X Font      ascent = 8 descent=-2
   Max Font    ascent = 8 descent=-2
 */
-const u8g_fntpgm_uint8_t ISO10646_1_5x7[1302] U8G_FONT_SECTION(".progmem.ISO10646_1_5x7") = {
-  0x00,0x06,0x0a,0x00,0xfe,0x07,0x02,0x22,0x03,0xb1,0x01,0x7f,0xff,0x08,0xfe,0x07,
-  0xff,0x05,0x08,0x08,0x06,0x00,0x00,0x40,0xf0,0xc8,0x88,0x88,0x98,0x78,0x10,0x05,
+#include <clib/u8g.h>
```
Comment (thinkyhead) [inline Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: @p3p, @yhfudev — If `dogm_font_data_ISO10646_1.h` is included as (or from) C++ then is this #include in need of a change? I also notice that this include line appears two times in this file. Here, and then again on line 138. Maybe it only needs to be included once.

File: Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h [hunk 2]
```diff
@@ -48,95 +49,96 @@ const u8g_fntpgm_uint8_t ISO10646_1_5x7[1302] U8G_FONT_SECTION(".progmem.ISO1064
...
+  0x05,0x05,0x05,0x06,0x00,0x00,0x88,0x88,0x88,0x50,0x20,0x05,0x05,0x05,0x06,0x00,
+  0x00,0x88,0x88,0xa8,0xa8,0x50,0x05,0x05,0x05,0x06,0x00,0x00,0x88,0x50,0x20,0x50,
+  0x88,0x05,0x07,0x07,0x06,0x00,0xfe,0x88,0x88,0x88,0x50,0x20,0x40,0x80,0x05,0x05,
+  0x05,0x06,0x00,0x00,0xf8,0x10,0x20,0x40,0xf8,0x03,0x09,0x09,0x06,0x01,0xff,0x20,
+  0x40,0x40,0x40,0x80,0x40,0x40,0x40,0x20,0x01,0x09,0x09,0x06,0x02,0xff,0x80,0x80,
+  0x80,0x80,0x80,0x80,0x80,0x80,0x80,0x03,0x09,0x09,0x06,0x01,0xff,0x80,0x40,0x40,
+  0x40,0x20,0x40,0x40,0x40,0x80,0x05,0x03,0x03,0x06,0x00,0x02,0x48,0xa8,0x90,0xff
+  };
 #else
   // extended (original) font (sysmbols 1 - 255)
```
Comment (thinkyhead) [inline Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: "symbols"

File: Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h [hunk 3]
```diff
@@ -26,19 +26,20 @@
   // reduced font (only sysmbols 1 - 127) - saves about 1278 bytes of FLASH
```
Comment (thinkyhead) [inline Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: "symbols"

File: Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h (hunk 1 above)
Comment (p3p) [reply Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: we should only use `U8glib.h` I removed the need to bypass that header, it caused no end of issues

Comment (yhfudev) [reply Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: It's already use U8glib.h after the merge with latest 2.0.x. see above

Comment (p3p) [reply Marlin/src/lcd/dogm/dogm_font_data_ISO10646_1.h]: on line 39 and 138 of this file it is still `<clib/u8g.h>`, that is what thinkyhead was referring to

**Comment by thinkyhead** [comment]
I've applied the noted changes. Also, seeing if it's ok to remove the `find` lines from `.travis.yml` which I assume were there for test purposes. Same with the "expose the C API" lines.

**Comment by thinkyhead** [comment]
I took the opportunity to tweak the shell scripts a bit to make them a little cleaner and to produce more "Marlin-like" final output.

I found that the `DN_EXEC` script path generated within `uxggenpages.sh` was coming out awkwardly —with triple-slashes at the end— although they seemed to work. I commented the code that was responsible for adding the extraneous slashes, so please test with your platform to see that the scripts still work.

**Comment by yhfudev** [comment]
@thinkyhead 
Thank you for cleaning the code.

**Comment by thinkyhead** [comment]
I've also added `get-bdf2u8g.sh` to automate the process of fetching, patching, and building `bdf2u8g`. That made it easier to explain the process in `uxggenpages.md`.

**Comment by rmoravcik** [comment]
Thank you guys :-)

**Pull request merged**